packages/node-core/src/indexer/connectionPoolState.manager.ts

```diff
@@ -79,9 +86,15 @@ export class ConnectionPoolStateManager<T extends IApiConnectionSpecific<any, an
       return primaryIndex;
     }
 
-    const indices = Object.keys(this.pool)
+    const initedIndices = Object.keys(this.pool)
       .map(Number)
-      .filter((index) => !this.pool[index].backoffDelay && this.pool[index].connected);
+      .filter((index) => !this.pool[index].initFailed);
+
+    if (initedIndices.length === 0) {
+      throw new Error(`Initialization failed for all endpoints. Please add healthier endpoints.`);
```

Comment:
This doesn't seem like the right place to throw this error. This isn't an init function